adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -576,32 +594,67 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     ))
   }
 
-  def loadVcf(filePath: String, sd: Option[SequenceDictionary]): RDD[VariantContext] = {
+  /**
+   * Loads a VCF file into an RDD.
+   *
+   * @param filePath The file to load.
+   * @param sdOpt An optional sequence dictionary, in case the sequence info
+   *              is not included in the VCF.
+   * @return Returns a VariantContextRDD.
+   */
+  def loadVcf(filePath: String,
+              sdOpt: Option[SequenceDictionary] = None): VariantContextRDD = {
+
+    // load vcf data
     val job = HadoopUtil.newJob(sc)
-    val vcc = new VariantContextConverter(sd)
     job.getConfiguration().set("io.compression.codecs", classOf[BGZFCodec].getCanonicalName())
+    val vcc = new VariantContextConverter(sdOpt)
     val records = sc.newAPIHadoopFile(
       filePath,
       classOf[VCFInputFormat], classOf[LongWritable], classOf[VariantContextWritable],
       ContextUtil.getConfiguration(job)
     )
+
+    // attach instrumentation
     if (Metrics.isRecording) records.instrument() else records
 
-    records.flatMap(p => vcc.convert(p._2.get))
+    // load vcf metadata
+    val (vcfSd, samples) = loadVcfMetadata(filePath)
```

Comment:
the header has already been read via `VCFInputFormat` above, and `VCFRecordReader` holds a reference to it.  Is there some way to pull that out?  `VCFHeaderReader` doesn't contain the decompression bits that `VCFRecordReader` does.